File: Java_Codes/6. Collections Framework/Collections/ListIteratorExample.java
package Collections;

import java.util.ArrayList;
import java.util.ListIterator;

public class ListIteratorExample {

    // Function to demonstrate ListIterator operations
    public static void listIteratorOperations() {
        // Create an ArrayList
        ArrayList<String> list = new ArrayList<>();
        list.add("Apple");
        list.add("Banana");
        list.add("Cherry");

        // Get the ListIterator
        ListIterator<String> listIterator = list.listIterator();

        // Iterate forward through the list
        System.out.println("Iterating forward through the list:");
        while (listIterator.hasNext()) {
            String element = listIterator.next();
            System.out.println(element);
        }

        // Iterate backward through the list
        System.out.println("Iterating backward through the list:");
        while (listIterator.hasPrevious()) {
            String element = listIterator.previous();
            System.out.println(element);
        }

        // Modify elements during iteration
        listIterator = list.listIterator(); // Reset the iterator
        while (listIterator.hasNext()) {
            String element = listIterator.next();
            if (element.equals("Banana")) {
                listIterator.set("Blueberry");
            }
        }

        // Display the list after modification
        System.out.println("List after modification: " + list);
    }

    public static void main(String[] args) {
        // Call the function to demonstrate ListIterator operations
        listIteratorOperations();
    }
}

File: Projects/PythonProject/Collections_Framework/Collections/ListIteratorExample.py
def list_iterator_operations():
    # Create a list
    fruits = ["Apple", "Banana", "Cherry"]

    # Iterate forward through the list
    print("Iterating forward through the list:")
    for fruit in fruits:
        print(fruit)

    # Iterate backward through the list
    print("Iterating backward through the list:")
    for fruit in reversed(fruits):
        print(fruit)

    # Modify elements during iteration
    for i in range(len(fruits)):
        if fruits[i] == "Banana":
            fruits[i] = "Blueberry"

    # Display the list after modification
    print("List after modification:", fruits)


if __name__ == "__main__":
    list_iterator_operations()
